### PR TITLE
Using a single Connection for CloudBigtableConnectionPool.

### PR DESCRIPTION
**Pull request by sduskis**
Moving the BigtableBufferedMutator offer() method.  This should also fix some of the issues in #630.  Each Connection opens up multiple channgles

### CODE REVIEW

File: bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableConnectionPool.java [hunk 1]
```diff
@@ -113,46 +76,20 @@ public PoolEntry getConnection(Configuration config) throws IOException {
...
-  }
-
-  public synchronized void returnConnection(PoolEntry entry) {
-    if (entry.isExpired()) {
-      closeAsynchronously(entry);
-    } else {
-      connections.put(entry.getKey(), entry);
-    }
+  @SuppressWarnings("unused")
+  protected Connection createConnection(Configuration config, String key) throws IOException {
```
Comment (kevinsi4508) [inline bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableConnectionPool.java]: remove "sting key"?

File: bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableConnectionPool.java [hunk 2]
```diff
@@ -51,21 +44,10 @@
   public static class PoolEntry {
     private final String key;
```
Comment (kevinsi4508) [inline bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableConnectionPool.java]: Do we still need "key" or "PoolEntry" itself?

**Comment by sduskis** [comment]
I removed PoolEntry.  I removed the key variable from createConnection.  I kept createConnection so that the test can override it.

**Comment by kevinsi4508** [comment]
LGTM

**Pull request merged**